src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/TypeNameHelper.cs

```diff
@@ -1,83 +1,96 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-#nullable disable
-
 using System;
-using System.Collections.Generic;
-using System.Text;
+using System.Collections.Immutable;
 using Microsoft.AspNetCore.Razor.Language.CodeGeneration;
 
 namespace Microsoft.AspNetCore.Razor.Language;
-internal class TypeNameHelper
+
+internal static class TypeNameHelper
 {
-    private static readonly string[] PredefinedTypeNames = new[] {
+    private const string GlobalPrefix = "global::";
+
+    private static readonly ImmutableArray<string> PredefinedTypeNames = new[]
+    {
         "bool",
         "int",
-        "uint",
-        "nint",
+        "string",
+        "float",
+        "double",
+        "decimal",
         "byte",
-        "char",
-        "long",
-        "ulong",
         "short",
-        "nuint",
-        "sbyte",
-        "float",
+        "long",
+        "char",
         "object",
-        "string",
+        "dynamic",
+        "uint",
         "ushort",
-        "double",
-        "decimal",
-        "dynamic"
-    };
+        "ulong",
+        "sbyte",
+        "nint",
+        "nuint",
+    }.ToImmutableArray();
 
-    public static void WriteGloballyQualifiedName(CodeWriter codeWriter, string typeName)
+    internal static string GetGloballyQualifiedNameIfNeeded(string typeName)
```

Comment:
Meta point: I really wonder if we need to make this an API in the compiler long term. There is so much language knowledge that needs to go into this code. Feels like something the compiler could help us out with.

Reply:
Meta point: You should really clarify the compiler you mean, since I think you're talking about the Roslyn compiler, but you're doing it from within the Razor compiler. :smile: